marshal/claim.go

```diff
@@ -422,6 +426,15 @@ func (c *claim) updateCurrentOffsets() (bool, int64) {
 	return didAdvance, c.offsets.Current
 }
 
+// heartbeatExpired returns whether or not our last successful heartbeat is so
+// long ago that we know we're expired.
+func (c *claim) heartbeatExpired() bool {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	return c.lastHeartbeat < time.Now().Unix()-HeartbeatInterval
```

Comment:
I believe `consumer unhealthy by heartbeat test` is unrelated as discussed f2f.

The point about the exact border is a good one. That definitely leads to this causing false positives. I'll update.